plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -2077,14 +2096,33 @@ function VocabBuilder:onWordLookedUp(word, title, is_manual)
             highlight = util.cleanupSelectedText(self.ui.highlight.selected_text.text)
         end
     end
-    DB:insertOrUpdate({
+
+    local update = function() DB:insertOrUpdate({
         book_title = title,
         time = os.time(),
         word = word,
         prev_context = prev_context,
         next_context = next_context,
         highlight = highlight ~= word and highlight or nil
-    })
+    }) end
+
+    local item = DB:hasWord(word)
+    if item then
+        local dialog = WordInfoDialog:new{
+            title = _("Vocabulary exists:") .. " " .. word,
+            highlighted_word = item.highlight or word,
+            book_title = item.book_title,
+            dates = _("Added on") .. " " .. os.date("%Y-%m-%d", item.create_time) .. " | " ..
+            _("Review scheduled at") .. " " .. os.date("%Y-%m-%d %H:%M", item.due_time),
```

Comment:
Preferably make that `T("Added on %1", os.date(etc))` and the same for the other one to give translators the freedom to make it the equivalent of for example `Added %1 on` if that's how it works in that language.

Reply:
Oh, I didn't mean to get rid of the `.. " | " ..` since that's just a visual decoration. :-)

Though since a phrase like "Review scheduled at" seems rather unlikely to show up elsewhere I suppose it doesn't matter much.

Reply:
Taking into consideration RTL scripts such as arabic, having them as one string seems to make more sense.

Reply:
I'm not sure I see your point. You're saying RTL would want to invert it back to LTR?

Reply:
Never mind, I didn't think it through. So do we split them into two parts each with a template or only use template on the first part?

Reply:
I'd make it two separate parts with each a template.

Reply:
Done again :)